src/flake8_quotes/rules.rs

```diff
@@ -68,54 +72,118 @@ pub fn quotes(
     };
 
     if is_docstring {
-        if raw_text.contains(good_docstring(&settings.docstring_quotes)) {
+        if raw_text.contains(good_docstring(&quotes_settings.docstring_quotes)) {
             return None;
         }
 
-        Some(Diagnostic::new(
-            violations::BadQuotesDocstring(settings.docstring_quotes.clone()),
+        let mut diagnostic = Diagnostic::new(
+            violations::BadQuotesDocstring(quotes_settings.docstring_quotes.clone()),
             Range::new(start, end),
-        ))
+        );
+        if matches!(autofix, flags::Autofix::Enabled) && settings.fixable.contains(&RuleCode::Q002)
+        {
+            let quote_count = if is_multiline { 3 } else { 1 };
+            let string_contents = &raw_text[quote_count..raw_text.len() - quote_count];
+            let quote = good_docstring(&quotes_settings.docstring_quotes).repeat(quote_count);
+            let new_string = format!("{prefix}{quote}{string_contents}{quote}");
```

Comment:
Do we need to handle escapes here? E.g., when converting this to double quotes:

```py
'Here's my content " with a double quote in the middle'
```

I'm guessing _no_, because we'd actually consider that acceptable (and thus not needing to be changed) based on the existing logic.